### PR TITLE
[Feature][OssFile Connector] Make Oss implement source factory and sink factory

### PR DESCRIPTION
**Pull request by EricJoy2048**
### Purpose of this pull request

1. This pr is make ossfile connector implement source factory and sink factory
2. Make OssFile Connector support Multi table reading and multi table writing.

### Does this PR introduce _any_ user-facing change?

Yes. For specific information, please refer to the changes in the document.

### How was this patch tested?

I tested in my local and all test case is success.
image

Because there is no available OSS environment, I disabled the test cases.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  3. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  4. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-examples/seatunnel-engine-examples/pom.xml [hunk 1]
```diff
@@ -29,6 +29,9 @@
 
     <properties>
         <hadoop3.version>3.1.4</hadoop3.version>
+        <hadoop-aliyun.version>3.1.4</hadoop-aliyun.version>
```
Comment (Carl-Zhou-CN) [inline seatunnel-examples/seatunnel-engine-examples/pom.xml]: Whether it needs to appear in example？

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-oss-e2e/src/test/resources/parquet/e2e.parquet
Comment (Carl-Zhou-CN) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-oss-e2e/src/test/resources/parquet/e2e.parquet:1]: Based on #5545 , we need to remove the binary and automatically generate it

File: docs/en/connector-v2/sink/OssFile.md [hunk 2]
```diff
@@ -362,6 +362,123 @@ sink {
...
 ```
 
+### Multiple Table
+
+For extract source metadata from upstream, you can use `${database_name}`, `${table_name}` and `${schema_name}` in the path.
+
+```bash
+
+env {
+  execution.parallelism = 1
```
Comment (hailin0) [inline docs/en/connector-v2/sink/OssFile.md]:
```suggestion
  parallelism = 1
```

File: docs/en/connector-v2/source/OssFile.md [hunk 3]
```diff
@@ -293,6 +293,177 @@ sink {
 }
 ```
 
+### Multiple Table
+
+No need to config schema file type, eg: `orc`.
+
+```
+env {
+  execution.parallelism = 1
```
Comment (hailin0) [inline docs/en/connector-v2/source/OssFile.md]:
```suggestion
  parallelism = 1
```

File: seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/local/sink/LocalFileSinkFactory.java [hunk 4]
```diff
@@ -17,32 +17,25 @@
...
 import org.apache.seatunnel.connectors.seatunnel.file.sink.commit.FileCommitInfo;
 import org.apache.seatunnel.connectors.seatunnel.file.sink.state.FileSinkState;
 
 import com.google.auto.service.AutoService;
 
 @AutoService(Factory.class)
-public class LocalFileSinkFactory
-        implements TableSinkFactory<
-                SeaTunnelRow, FileSinkState, FileCommitInfo, FileAggregatedCommitInfo> {
+public class LocalFileSinkFactory extends BaseMultipleTableFinkSinkFactory {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/local/sink/LocalFileSinkFactory.java:38]: Is this change related to oss?

File: seatunnel-dist/pom.xml [hunk 5]
```diff
@@ -636,12 +636,6 @@
                 </dependency>
 
                 <!-- hadoop jar -->
-                <dependency>
-                    <groupId>org.apache.hadoop</groupId>
-                    <artifactId>hadoop-aliyun</artifactId>
-                    <version>${hadoop-aliyun.version}</version>
-                    <scope>provided</scope>
-                </dependency>
```
Comment (hailin0) [inline seatunnel-dist/pom.xml:644]: why delete it

File: docs/en/connector-v2/sink/OssFile.md (hunk 2 above)
Comment (hailin0) [reply docs/en/connector-v2/sink/OssFile.md]: https://github.com/apache/seatunnel/pull/6003

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-oss-e2e/src/test/resources/excel/fake_to_oss_excel.conf [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+env {
+  execution.parallelism = 1
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-oss-e2e/src/test/resources/excel/fake_to_oss_excel.conf]:
```suggestion
  parallelism = 1
```

change all execution.parallelism to parallelism
https://github.com/apache/seatunnel/pull/6003

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-oss-e2e/src/test/resources/parquet/e2e.parquet
Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-oss-e2e/src/test/resources/parquet/e2e.parquet:1]: > Based on #5545 , we need to remove the binary and automatically generate it

I asked other open source projects, but this is not mandatory.

File: seatunnel-dist/pom.xml (hunk 5 above)
Comment (EricJoy2048) [reply seatunnel-dist/pom.xml:644]: Because it's repetitive. In the line 665 we already have `hadoop-aliyun`.

File: seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/local/sink/LocalFileSinkFactory.java (hunk 4 above)
Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/local/sink/LocalFileSinkFactory.java:38]: > Is this change related to oss?

While adding multi-table support for oss, I found that `BaseMultipleTableFinkSinkFactory` could be added to make the code cleaner, so I needed to modify the `LocalFileSinkFactory` implementation

File: seatunnel-examples/seatunnel-engine-examples/pom.xml (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-examples/seatunnel-engine-examples/pom.xml]: > Whether it needs to appear in example？

Thanks, I will roll back it.

File: seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/local/sink/LocalFileSinkFactory.java (hunk 4 above)
Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/local/sink/LocalFileSinkFactory.java:38]: > Is this change related to oss?

While adding multi-table support for oss, I found that `BaseMultipleTableFinkSinkFactory` could be added to make the code cleaner, so I needed to modify the `LocalFileSinkFactory` implementation

File: seatunnel-examples/seatunnel-engine-examples/pom.xml (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-examples/seatunnel-engine-examples/pom.xml]: done

**Review by Carl-Zhou-CN — APPROVED** [review]
LGTM

**Review by TaoZex — APPROVED** [review]
+1

**Review by ruanwenjun — APPROVED** [review]
+1
